src/backend/distributed/commands/call.c

```diff
@@ -217,3 +202,83 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 
 	return true;
 }
+
+
+/*
+ * ShardPlacementWhenColocatedWithReferenceTable decides on a placement for delegating
+ * a procedure call that reads from a reference table.
+ *
+ * If citus.task_assignment_policy is set to round-robin, we assign a different placement
+ * on consecutive runs. Otherwise the function returns the first placement available.
+ */
+static ShardPlacement *
+ShardPlacementWhenColocatedWithReferenceTable(CitusTableCacheEntry *cacheEntry)
+{
+	const ShardInterval *shardInterval = cacheEntry->sortedShardIntervalArray[0];
+	const uint64 referenceTableShardId = shardInterval->shardId;
+	List *placementList = ActiveShardPlacementList(referenceTableShardId);
```

Comment:
I'm wondering what we should do if the coordinator is in the metadata. It would probably be helpful to at least be consistent. Whether we'll use the coordinator currently depends on the placement ID order, which is somewhat arbitrary (depending on when the coordinator was added or the table was created).

For read-only router queries, we end up always querying the coordinator, mainly because it is in the placement list and we always do local placements before remote placements (and then never do remote placements). 

In general, ActiveShardPlacementList seems a bit of a mess. I wonder whether we should order by groupid to make sure we always access nodes in the same order with the coordinator coming first.

Reply:
I introduced a new comparison function `CompareShardPlacementsByGroupId`. Here is the behaviour:

- If round-robin is not used:
 (i) and we have coordinator in the metadata, we will always run the functions/procedures on the coordinator
 (ii) we do not have coordinator in the metadata, the delegation will be done in the first added worker that is still active.

Maybe, we can change the behavior of `ActiveShardPlacementList` completely. However this would definitely be out of the scope of this PR.

Reply:
CompareShardPlacementsByGroupId makes sense. Thinking about it more. What if we always exclude the coordinator? The user clearly wanted to delegate this function call. It would be weird to still call it on the coordinator regardless of the task-assignment policy.

Reply:
I remove all placements on the coordinator, if there exists some placements that actually are not in the coordinator.

This means that if the user has no active workers, the distributed function will be delegated to the coordinator. I thought this would be helpful for starter plan users that want to try out if distributed functions work for them.